File: 03-03/search/linear/linear.c
#include <stdio.h>

int linear_search(int array[], int size, int target) {
    for (int i = 0; i < size; i++) {
        if (array[i] == target) {
            return i;
        }
    }
    return -1;
}

int main() {
    int array[] = {1, 2, 3, 4, 5};
    int size = sizeof(array) / sizeof(array[0]);
    int target;

    printf("Enter the number to search: ");
    scanf("%d", &target);

    int result = linear_search(array, size, target);

    if (result == -1) {
        printf("Element not found\n");
    } else {
        printf("Element found at index %d\n", result);
    }

    return 0;
}

File: 03-03/search/linear/linear.py
def linear_search(array, target):
    for index, value in enumerate(array):
        if value == target:
            return index
    return -1

def main():
    array = [1, 2, 3, 4, 5]
    target = int(input("Enter the number you want to search: "))
    result = linear_search(array, target)
    
    if result == -1:
        print("Element not found!")
    else:
        print(f"Element found at index {result}")
        
if __name__ == "__main__":
    main()